### DISCUSSION TITLE
Hey, you did it and you did it great :)

### DISCUSSION THREAD

**Discussion by matthias882** [omerfaruk-aran/esphome_samsung_hvac_bus#68] [Show and tell]
One year ago i startet to tried to control my ac over the indoor/outdoor databus with more or less success. After a while and after some not so bad results i had no more time and to less knowledge to bring this project to a good end...
Now one year ist gone and i thougt in the last days it would be great go on in this project. But then i remembered you and I thougt to myself, lets see how his Project is grown in this time. And what should i say, its great!!!
Your component can generate one climate-instance for each indoor unit. How many hours I tried this without success :)
It reads out the data of the indoor units fine and the code has a much better structure than my quick and dirty code.

Thats my excel-sheet where I sniffed my non-nasa communication. I assigned to all the commands on my bus the values and and verified them by watching the values in different operation modes and parallel in SNet-Software.
[Commands_F1F2_Com_.xlsx](https://github.com/lanwin/esphome_samsung_ac/files/14116764/Commands_F1F2_Com_.xlsx)
It would be great if you can use it to include some nice information like the outdoor-temperature or the power consumption of the outdoor-unit.

Currently the Component doesn't control anything in my system, it only reads values. there seems to be an issue in the non-nasa communication. No data ist sent over the uart-port when i try to set some values. But I see you are working currently on this section every day. If i can help you for non-nasa device testing this would be no problem.

Thanks for the great code so far :)
matthias882

**Comment by lanwin** [comment]
Hi, its good to hear that you are OK! 

And I have to really say thank you very very much!!! I didnt know if I had started working on this, when I wouldn't had your code as base to start.

Thank you for that Excel that is also a great help. 

Please join us on this thread https://github.com/lanwin/esphome_samsung_ac/issues/1 . 

I figured out that you can only control it when you send your message right after the c6 command. But it seems that this message did not always get sent. May its required to enable it first. This could be the reason why it worked after he started SNET https://github.com/lanwin/esphome_samsung_ac/issues/1#issuecomment-1918994659 .

Could you please enable debug_log_messages [https://github.com/lanwin/esphome_samsung_ac and ](https://github.com/lanwin/esphome_samsung_ac?tab=readme-ov-file#development) and then check if you see any c6 cmd?
